{"identifier":{"url":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/RetryPolicy\/getRetryWaitTime(error:attempt:)","interfaceLanguage":"swift"},"metadata":{"role":"symbol","title":"getRetryWaitTime(error:attempt:)","symbolKind":"method","modules":[{"name":"SotoCore"}],"fragments":[{"kind":"keyword","text":"func"},{"text":" ","kind":"text"},{"text":"getRetryWaitTime","kind":"identifier"},{"text":"(","kind":"text"},{"kind":"externalParam","text":"error"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"Error","preciseIdentifier":"s:s5ErrorP"},{"kind":"text","text":", "},{"kind":"externalParam","text":"attempt"},{"kind":"text","text":": "},{"kind":"typeIdentifier","preciseIdentifier":"s:Si","text":"Int"},{"text":") -> ","kind":"text"},{"preciseIdentifier":"s:8SotoCore11RetryStatusO","kind":"typeIdentifier","text":"RetryStatus"},{"text":"?","kind":"text"}],"roleHeading":"Instance Method","required":true,"externalID":"s:8SotoCore11RetryPolicyP03getC8WaitTime5error7attemptAA0C6StatusOSgs5Error_p_SitF"},"kind":"symbol","schemaVersion":{"major":0,"minor":3,"patch":0},"abstract":[{"type":"text","text":"Returns whether we should retry and how long we should wait before retrying"}],"primaryContentSections":[{"kind":"declarations","declarations":[{"platforms":["macOS"],"tokens":[{"kind":"keyword","text":"func"},{"kind":"text","text":" "},{"text":"getRetryWaitTime","kind":"identifier"},{"kind":"text","text":"("},{"kind":"externalParam","text":"error"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"Error","preciseIdentifier":"s:s5ErrorP"},{"text":", ","kind":"text"},{"text":"attempt","kind":"externalParam"},{"text":": ","kind":"text"},{"text":"Int","kind":"typeIdentifier","preciseIdentifier":"s:Si"},{"text":") -> ","kind":"text"},{"preciseIdentifier":"s:8SotoCore11RetryStatusO","text":"RetryStatus","kind":"typeIdentifier","identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/RetryStatus"},{"kind":"text","text":"?"}],"languages":["swift"]}]},{"parameters":[{"content":[{"inlineContent":[{"text":"Error returned by HTTP client","type":"text"}],"type":"paragraph"}],"name":"error"},{"content":[{"inlineContent":[{"type":"text","text":"Retry attempt number"}],"type":"paragraph"}],"name":"attempt"}],"kind":"parameters"}],"sections":[],"variants":[{"paths":["\/documentation\/sotocore\/retrypolicy\/getretrywaittime(error:attempt:)"],"traits":[{"interfaceLanguage":"swift"}]}],"hierarchy":{"paths":[["doc:\/\/codes.soto.soto-core\/documentation\/SotoCore","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/RetryPolicy"]]},"references":{"doc://codes.soto.soto-core/documentation/SotoCore/RetryPolicy":{"url":"\/documentation\/sotocore\/retrypolicy","role":"symbol","fragments":[{"text":"protocol","kind":"keyword"},{"text":" ","kind":"text"},{"kind":"identifier","text":"RetryPolicy"}],"title":"RetryPolicy","navigatorTitle":[{"text":"RetryPolicy","kind":"identifier"}],"type":"topic","kind":"symbol","identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/RetryPolicy","abstract":[{"type":"text","text":"Protocol for Retry strategy. Has function returning amount of time before the next retry after an HTTP error"}]},"doc://codes.soto.soto-core/documentation/SotoCore/RetryPolicy/getRetryWaitTime(error:attempt:)":{"kind":"symbol","role":"symbol","title":"getRetryWaitTime(error:attempt:)","abstract":[{"type":"text","text":"Returns whether we should retry and how long we should wait before retrying"}],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/RetryPolicy\/getRetryWaitTime(error:attempt:)","fragments":[{"text":"func","kind":"keyword"},{"text":" ","kind":"text"},{"text":"getRetryWaitTime","kind":"identifier"},{"kind":"text","text":"("},{"kind":"externalParam","text":"error"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"Error","preciseIdentifier":"s:s5ErrorP"},{"kind":"text","text":", "},{"text":"attempt","kind":"externalParam"},{"kind":"text","text":": "},{"text":"Int","kind":"typeIdentifier","preciseIdentifier":"s:Si"},{"text":") -> ","kind":"text"},{"kind":"typeIdentifier","preciseIdentifier":"s:8SotoCore11RetryStatusO","text":"RetryStatus"},{"kind":"text","text":"?"}],"url":"\/documentation\/sotocore\/retrypolicy\/getretrywaittime(error:attempt:)","type":"topic","required":true},"doc://codes.soto.soto-core/documentation/SotoCore/RetryStatus":{"title":"RetryStatus","kind":"symbol","url":"\/documentation\/sotocore\/retrystatus","abstract":[{"text":"Return value for ","type":"text"},{"type":"codeVoice","code":"RetryPolicy.getRetryWaitTime"},{"type":"text","text":". Either retry after time amount or donâ€™t retry"}],"type":"topic","identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/RetryStatus","fragments":[{"kind":"keyword","text":"enum"},{"kind":"text","text":" "},{"kind":"identifier","text":"RetryStatus"}],"navigatorTitle":[{"kind":"identifier","text":"RetryStatus"}],"role":"symbol"},"doc://codes.soto.soto-core/documentation/SotoCore":{"abstract":[{"text":"The core framework for Soto, a Swift SDK for AWS","type":"text"}],"title":"SotoCore","identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore","url":"\/documentation\/sotocore","kind":"symbol","type":"topic","role":"collection"}}}